### ISSUE TITLE
Global media buttons stopped working

### ISSUE THREAD

**Issue by xmbwd** [pithos/pithos#447]
I am on Ubuntu 16.04 and used Pithos for over a year up until a few months ago.  During that time the global media keys worked as expected.  I moved to Chrome for a few months and assigned the media keys (globally) to Chrome for media playback.  I then removed these global re-assignments and went back to Pithos.  

On return to Pithos, the media keys no longer worked.  I uninstalled Chrome and checked other media players (e.g., Tomahawk, MPV, etc.) and the media keys worked fine.  I also tried `xev`, and the media keys are appropriately reporting `XF86AudioPlay`, `XF86AudioNext`, etc.

Here is the output from `pithos -v`:
```
INFO - application:do_activate:169 - Pithos 1.3.0
INFO - pithos:init_core:329 - Created temporary directory /tmp/pithos-cfu8v0es
INFO - plugin:enable:39 - Enabling module notification_icon
INFO - plugin:enable:39 - Enabling module notify
INFO - plugin:enable:39 - Enabling module journald_logging
INFO - journald_logging:_on_logging_changed:54 - setting journald logging level to: verbose
INFO - plugin:enable:39 - Enabling module mediakeys
INFO - mediakeys:bind_dbus:47 - Got session bus
INFO - mediakeys:bind_dbus:98 - Bound media keys with DBUS org.gnome.SettingsDaemon.MediaKeys
INFO - pandora:connect:290 - Time offset is 1.8524224758148193
INFO - pithos:pandora_ready:592 - Pandora connected
INFO - pithos:process_stations:647 - Restoring saved station: id = 1591875593442521234
INFO - pithos:station_changed:901 - Selecting station 1591875593442521234; total = 32
INFO - pandora:get_playlist:392 - pandora: Get Playlist
INFO - pithos:start_song:690 - Starting song: index = 0
INFO - pandora:audioUrl:505 - Using audio quality mediumQuality: 64 aacplus
INFO - pithos:art_callback:823 - Downloaded album art for 0
INFO - pithos:art_callback:823 - Downloaded album art for 1
INFO - pithos:art_callback:823 - Downloaded album art for 2
INFO - pithos:art_callback:823 - Downloaded album art for 3
INFO - pithos:check_if_song_is_ad:981 - Not an Ad..
```

Also, I checked and `gir1.2-keybinder-3.0` is installed.  Any idea why Pithos is not responding to the media keys -- even though it appears to be binding them?

**Comment by JasonLG1979** [comment]
Are using the PPA or the master git branch? If you're using the PPA could you try master? @TingPing has made a few changes in the media keys plugin since the release of 1.3.0.

**Comment by JasonLG1979** [comment]
> Also, I checked and gir1.2-keybinder-3.0

That shouldn't matter. Your keys are *supposed* have been bound via DBus so that's not actually even being used. Keybinder is just a fallback.

**Comment by xmbwd** [comment]
Yep -- the PPA.  Let me switch and report back.

**Comment by JasonLG1979** [comment]
I'll fire up a Ubuntu 16.04 VM and see what I come up with.

**Comment by xmbwd** [comment]
I installed the master, but the problem persisted.  Where is the user config?  The PPA uninstall and master install kept my old settings.   I want to delete it and see if that helps.

**Comment by JasonLG1979** [comment]
> I installed the master, but the problem persisted. Where is the user config? The PPA uninstall and master install kept my old settings. I want to delete it and see if that helps.

That wouldn't do anything. The only setting for the Media Keys plugin is "enabled" which just keeps track of it's enabled or not. The plugin binds to the generic media keys. That's not configurable.

**Comment by JasonLG1979** [comment]
As soon as I get that Ubuntu VM setup(installing now) I'll see if it's something screwy with Ubuntu or not.

**Comment by JasonLG1979** [comment]
I have git master installed on a brand new clean and updated Ubuntu 16.04 VM with nothing changed except d-feet and Pithos and it's dependencies installed. I can not reproduce your issue @xmbwd. All the media keys work as expected.

**Comment by xmbwd** [comment]
Ok.  Not sure what to do then.  I can close the thread.  

Do you offhand know of another app that uses the same media key binding approach as Pithos?  I will test that.

**Comment by JasonLG1979** [comment]
Pithos doesn't take control of the keys. It listens for key presses of the media keys you set in your settings. You may need to configure your media keys.

Here's what mine looks like. My keyboard doesn't have fancy dedicated media keys.
![screenshot from 2017-04-17 14-13-04](https://cloud.githubusercontent.com/assets/6667703/25101069/6f46668c-2378-11e7-9032-b500c2177610.png)

Pithos also does not listen for volume. Your volume keys will adjust the system volume.

**Comment by xmbwd** [comment]
I set my FN keys to control media.  Here is what mine looks like:

![screenshot from 2017-04-17 12-18-58](https://cloud.githubusercontent.com/assets/12945044/25101198/22191cd4-2368-11e7-91cd-3de30351f003.png)

**Comment by JasonLG1979** [comment]
Your keys aren't set for anything. Click the right description and press the key you want to bind to that function to bind it.

**Comment by xmbwd** [comment]
`Audio Play` is what it says when I bind the key to the Play media key button (F8 on my keyboard).  If I remove it, it will say disabled. See below.  

![screenshot from 2017-04-17 12-25-41](https://cloud.githubusercontent.com/assets/12945044/25101412/0c61b99a-2369-11e7-995a-9bd347afab33.png)

**Comment by JasonLG1979** [comment]
Strange. I bound mine to the FN keys on my keyboard also and as you see mine just show F1 and s on.

**Comment by xmbwd** [comment]
Yeah, I think that's because you haven't set your keyboard up to switch the FN keys with media keys -- which you wouldn't on a VM you just downloaded.  Thanks though, not sure what's going on here.  

I did just test again to see if Chrome's media keys extension is the issue - turning it on/off - and again Tomahawk and MPV work when it is toggled, but Pithos does not.  No idea what is causing this.

**Comment by JasonLG1979** [comment]
> Yeah, I think that's because you haven't set your keyboard up to switch the FN keys with media keys --

I'm not sure how you'd do that? They may be your problem.

**Comment by xmbwd** [comment]
That is the problem!  It is set that way b/c I'm on a Mac using a Mac keyboard.  For some reason, Pithos only takes the Play command if it is F8, whereas other apps can figure out the media key `Audio Play`.  When set as below, it works in Pithos.  

![screenshot from 2017-04-17 12-37-33](https://cloud.githubusercontent.com/assets/12945044/25101839/abdc1316-236a-11e7-8354-597ce8153d60.png)

**Comment by JasonLG1979** [comment]
Good deal. glad you figured it out.

**Comment by xmbwd** [comment]
Kind of -- now I have to go back and figure out how to change the media keys to FN keys.  Thanks for working through it.

**Comment by xmbwd** [comment]
One more comment, in case anyone experiences the same issue.  I was able to "fix" it (so that the key command `XF86AudioPlay` pauses and plays in Piths) by resetting my media keys in `dconf editor`.  See the picture below.  I just clicked the defaults for each of the media keys.  My guess is that something with Google Chrome's media keys extensions messed with these settings, and they could only be reset manually. 

![screenshot from 2017-04-17 13-03-01](https://cloud.githubusercontent.com/assets/12945044/25102978/89ce25f2-236f-11e7-99e5-103b4b8f753e.png)
